### PR TITLE
Roselia Donator Items

### PR DESCRIPTION
**Pull request by IgiariValkyr**
## About The Pull Request

Donator items for ultimarifox involving some old Red Sec items and a plushie.
-Adds: "black and red turtleneck" variant of the head of security turtleneck, using the old sprite from times long gone
-Adds: "red-tinted giga HUD gar glasses" variant of the giga HUD gar glasses that use the old red coloration
-Adds: "obscene sergal plushie" which is literally just Roselia and her over-stuffing in plush form

## How This Contributes To The Skyrat Roleplay Experience

It doesn't at all. It's donator items. FLOOF.

## Changelog

:cl:
add: Added Donator Items for ultimarifox
/:cl:

### CODE REVIEW

File: modular_skyrat/modules/customization/game/objects/items/plushes.dm [hunk 1]
```diff
@@ -536,3 +536,13 @@
...
+
+/obj/item/toy/plush/roselia
+	name = "obscene sergal plushie"
+	desc = "A plush recreation of a pink sergal. The chest is extremely padded and the small plush clothes are barely holding themselves together."
+	icon = 'modular_skyrat/master_files/icons/obj/plushes.dmi'
+	icon_state = "plushie_roselia"
+	attack_verb_continuous = list("hugs", "gropes")
+	attack_verb_simple = list("hug", "grope")
+	squeak_override = list('modular_skyrat/modules/emotes/sound/voice/merp.ogg' = 1)
+	young = 1 //No.
```
Comment (Zonespace27) [inline modular_skyrat/modules/customization/game/objects/items/plushes.dm]: true/false defines

Comment (Zonespace27) [inline modular_skyrat/modules/customization/game/objects/items/plushes.dm]: EoL

File: modular_skyrat/modules/customization/modules/clothing/~donator/donator_clothing.dm [hunk 2]
```diff
@@ -1360,3 +1360,19 @@
...
+	icon_state = "hosaltred"
+	worn_icon = 'modular_skyrat/master_files/icons/donator/mob/clothing/uniform.dmi'
+	worn_icon_digi = null
+
+/obj/item/clothing/glasses/hud/security/sunglasses/gars/giga/roselia
+	name = "red-tinted giga HUD gar glasses"
+	desc = "GIGA GAR glasses with a security hud implanted in the lens. Reminds you of a time before the color blue."
+	icon = 'modular_skyrat/master_files/icons/donator/obj/clothing/glasses.dmi'
+	icon_state = "supergarsred"
+	worn_icon = 'modular_skyrat/master_files/icons/donator/mob/clothing/eyes.dmi'
```
Comment (Zonespace27) [inline modular_skyrat/modules/customization/modules/clothing/~donator/donator_clothing.dm]: EoL

**Review by Zonespace27 — CHANGES_REQUESTED** [review]

File: modular_skyrat/modules/customization/game/objects/items/plushes.dm [hunk 3]
```diff
@@ -536,3 +536,13 @@
...
 	mood_change = 0
 	timeout = 2 MINUTES
+
+/obj/item/toy/plush/roselia
+	name = "obscene sergal plushie"
+	desc = "A plush recreation of a pink sergal. The chest is extremely padded and the small plush clothes are barely holding themselves together."
+	icon = 'modular_skyrat/master_files/icons/obj/plushes.dmi'
+	icon_state = "plushie_roselia"
+	attack_verb_continuous = list("hugs", "gropes")
+	attack_verb_simple = list("hug", "grope")
```
Comment (GoldenAlpharex) [inline modular_skyrat/modules/customization/game/objects/items/plushes.dm]: I'd rather not deal with people groping others with plushies.
```suggestion
	attack_verb_continuous = list("hugs")
	attack_verb_simple = list("hug")
```

File: modular_skyrat/modules/loadouts/loadout_items/donator/personal/donator_personal.dm [hunk 4]
```diff
@@ -407,3 +407,22 @@
...
+/datum/loadout_item/glasses/redgigagar
+	name = "red-tinted giga HUD gar glasses"
+	item_path = /obj/item/clothing/glasses/hud/security/sunglasses/gars/giga/roselia
+	ckeywhitelist = list("ultimarifox")
+	restricted_roles = list(JOB_HEAD_OF_SECURITY, JOB_SECURITY_OFFICER)
+
+/datum/loadout_item/toys/roselia_plush
+	name = "obscene sergal plushie"
+	item_path = /obj/item/toy/plush/roselia
+	ckeywhitelist = list("ultimarifox")
```
Comment (GoldenAlpharex) [inline modular_skyrat/modules/loadouts/loadout_items/donator/personal/donator_personal.dm]: These names should be capitalized, like all the other loadout entries.

File: modular_skyrat/modules/loadouts/loadout_items/donator/personal/donator_personal.dm [hunk 5]
```diff
@@ -407,3 +407,22 @@
 	name = "Squish-Me-Tesh Plush"
 	item_path = /obj/item/toy/plush/donator/tesh
 	ckeywhitelist = list("tobjv")
+
+/datum/loadout_item/under/jumpsuit/redhosneck
+	name = "black and red turtleneck"
+	item_path = /obj/item/clothing/under/rank/security/head_of_security/alt/roselia
+	ckeywhitelist = list("ultimarifox")
+	restricted_roles = list(JOB_HEAD_OF_SECURITY, JOB_SECURITY_OFFICER)
```
Comment (GoldenAlpharex) [inline modular_skyrat/modules/loadouts/loadout_items/donator/personal/donator_personal.dm]: No HoS clothes for regular officers, thank you.
```suggestion
	restricted_roles = list(JOB_HEAD_OF_SECURITY)
```

**Review by GoldenAlpharex — CHANGES_REQUESTED** [review]

**Comment by IgiariValkyr** [comment]
👍 
Not sure what EoL means but the rest are fixed.

**Comment by Zonespace27** [comment]
> 👍 Not sure what EoL means but the rest are fixed.

Add a newline at the end of the file

**Comment by IgiariValkyr** [comment]
> > 👍 Not sure what EoL means but the rest are fixed.
> 
> Add a newline at the end of the file

👍

File: modular_skyrat/modules/customization/game/objects/items/plushes.dm [hunk 6]
```diff
@@ -536,3 +536,14 @@
...
+
+/obj/item/toy/plush/roselia
+	name = "obscene sergal plushie"
+	desc = "A plush recreation of a pink sergal. The chest is extremely padded and the small plush clothes are barely holding themselves together."
+	icon = 'modular_skyrat/master_files/icons/obj/plushes.dmi'
+	icon_state = "plushie_roselia"
+	attack_verb_continuous = list("hugs")
+	attack_verb_simple = list("hug")
+	squeak_override = list('modular_skyrat/modules/emotes/sound/voice/merp.ogg' = 1)
+	young = 1 //No.
```
Comment (Zonespace27) [inline modular_skyrat/modules/customization/game/objects/items/plushes.dm]: use true/false instead of 1/0

**Review by Zonespace27 — CHANGES_REQUESTED** [review]

File: modular_skyrat/modules/customization/game/objects/items/plushes.dm (hunk 6 above)
Comment (IgiariValkyr) [reply modular_skyrat/modules/customization/game/objects/items/plushes.dm]: oh. all the other plushies in the file use 1. or the ones at the top of the file at least

File: modular_skyrat/modules/customization/game/objects/items/plushes.dm [hunk 7]
```diff
@@ -536,3 +536,14 @@
...
+
+/obj/item/toy/plush/roselia
+	name = "obscene sergal plushie"
+	desc = "A plush recreation of a pink sergal. The chest is extremely padded and the small plush clothes are barely holding themselves together."
+	icon = 'modular_skyrat/master_files/icons/obj/plushes.dmi'
+	icon_state = "plushie_roselia"
+	attack_verb_continuous = list("hugs")
+	attack_verb_simple = list("hug")
+	squeak_override = list('modular_skyrat/modules/emotes/sound/voice/merp.ogg' = 1)
+	young = false //I don't even know what this does for plushies
```
Comment (Zonespace27) [inline modular_skyrat/modules/customization/game/objects/items/plushes.dm]:
```suggestion
	young = FALSE
```
it's a preprocessor argument, it's in caps.

**Pull request merged**